src/file.ts

```diff
@@ -2294,21 +2330,61 @@ class File extends ServiceObject<File> {
     const expiresInSeconds =
         Math.round(expiresInMSeconds / 1000);  // The API expects seconds.
 
-    const config: GetSignedUrlConfigInternal = Object.assign({}, cfg);
-
-    config.action = ({
-      read: 'GET',
-      write: 'PUT',
-      delete: 'DELETE',
-      resumable: 'POST',
-    } as {[index: string]: string})[config.action];
+    const method = ActionToHTTPMethod[cfg.action];
 
     const name = encodeURIComponent(this.name);
-    config.resource = '/' + this.bucket.name + '/' + name;
+    const resource = `/${this.bucket.name}/${name}`;
+
+    const version = cfg.version || DEFAULT_SIGNING_VERSION;
+
+    const config: GetSignedUrlConfigInternal = Object.assign({}, cfg, {
+      method,
+      expiration: expiresInSeconds,
+      resource,
+      name,
+    });
+
+    let promise: Promise<SignedUrlQuery>;
+    if (version === 'v2') {
```

Comment:
Could you log the future change:
> You have generated a signed URL using the default v2 signing implementation. In the future, this will default to v4. You may experience breaking changes if you use longer than 7 day expiration times with v4. To opt-in to the behavior specify config.version='v4'

Reply:
We don't really do these in Node. 

@JustinBeckwith do we have any best-practices/alternatives for warning users of impending change in API during runtime in addition to jsdoc notes?

Reply:
Yes we do!  Checkout what I did here for google-auth-library:
https://github.com/googleapis/google-auth-library-nodejs/blob/master/src/messages.ts

You want to use `process.emitWarning`, but there is a lot of nuance.  Following gal is probably a good idea.

Reply:
gal?

Reply:
Sorry, google-auth-library

Reply:
@frankyn Do we only emit warning if the user didn't provide the `version` config (and defaults to 'v2'), and not when the user specified specifically 'v2'?

Reply:
Only default is fine in this case.

Reply:
> Yes we do!

Heh I must've remembered a different conversion about emitting warnings in Node. 😛 
I am trying to add a similar emitWarning logic here and I feel like I'm duplicating most of the code you have on `google-auth-library`. Any hints here, or should I just go ahead with it?

Reply:
It's entirely possible there's a library to do this - not sure!  @googleapis/node-team?

Reply:
@JustinBeckwith are you okay with me keeping the (mostly duplicated) code? I've looked high and low and couldn't find anything :(

Reply:
I am ok with it